### PR TITLE
HVG-903: Show login when pressing login button in more options

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [x] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/androidTest/java/com/hedvig/app/feature/embark/MoreOptionsTest.kt [hunk 1]
```diff
@@ -46,4 +66,22 @@ class MoreOptionsTest : TestCase() {
...
+        val newInstance = MoreOptionsActivity.newInstance(context())
+        activityRule.launch(newInstance)
+
+        onScreen<MoreOptionsScreen> {
+            loginButton {
+                click()
+            }
+        }
+
+        onScreen<ZignSecScreen> {
```
Comment (hedvigoscar) [inline app/src/androidTest/java/com/hedvig/app/feature/embark/MoreOptionsTest.kt:81]: Avoid starting new activities in a test. Instead, prefer asserting on stubbed intents.

This is the general procedure for doing that:

1. Replace `LazyActivityScenarioRule` with `LazyIntentsActivityScenarioRule`
2. Create a `KIntent` in the `Screen` you're working with
3. Before performing the action that causes the intent to fire, run `yourKIntent { stub() }`
4. After performing the action that causes the intent to fire, assert that it was fired with `yourKIntent { intended() }`

File: app/src/androidTest/java/com/hedvig/app/feature/embark/marketselected/MarketSelectedScreen.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.hedvig.app.feature.embark.marketselected
+
+import com.agoda.kakao.screen.Screen
+import com.agoda.kakao.text.KButton
+import com.agoda.kakao.text.KTextView
+import com.hedvig.app.R
+
+class MarketSelectedScreen: Screen<MarketSelectedScreen>() {
```
Comment (hedvigoscar) [inline app/src/androidTest/java/com/hedvig/app/feature/embark/marketselected/MarketSelectedScreen.kt:8]:
```suggestion
object MarketSelectedScreen: KScreen<MarketSelectedScreen>() {
```

Also, don't forget to format your files 😉

File: app/src/androidTest/java/com/hedvig/app/feature/embark/screens/ZignSecScreen.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.hedvig.app.feature.embark.screens
+
+import com.agoda.kakao.common.views.KView
+import com.agoda.kakao.screen.Screen
+import com.hedvig.app.R
+
+class ZignSecScreen: Screen<ZignSecScreen>() {
```
Comment (hedvigoscar) [inline app/src/androidTest/java/com/hedvig/app/feature/embark/screens/ZignSecScreen.kt:7]: When moving to using stubbed intents, we no longer need this screen-file I think 🙌

File: app/src/main/java/com/hedvig/app/feature/embark/ui/MoreOptionsActivity.kt [hunk 4]
```diff
@@ -36,6 +39,10 @@ class MoreOptionsActivity : BaseActivity(R.layout.activity_more_options) {
 
             recycler.adapter = MoreOptionsAdapter(viewModel)
 
+            logIn.setOnClickListener {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/embark/ui/MoreOptionsActivity.kt:42]:
```suggestion
            logIn.setHapticClickListener {
```

**Review by hedvigoscar — CHANGES_REQUESTED** [review]
A few things to adress, but mostly good!

File: app/src/androidTest/java/com/hedvig/app/feature/embark/MoreOptionsTest.kt (hunk 1 above)
Comment (hugokallstrom) [reply app/src/androidTest/java/com/hedvig/app/feature/embark/MoreOptionsTest.kt:81]: Cool, thanks for clarifying! Will change

**Review by hedvigoscar — APPROVED** [review]

**Comment by hedvigoscar** [comment]
Although make sure that the upstream branch is rebased before merging 😧

**Pull request merged**